ios-target/View/Home/HomeViewController.swift

```diff
@@ -78,6 +78,28 @@ class HomeViewController: UIViewController {
       mapView.addOverlay(overlay)
     }
   }
+  
+  func removeTargetAnnotation(targetId: Int) {
+    viewModel.removeTarget(targetId: targetId)
+    
+    mapView.annotations.forEach {
+      if
+        let annotation = $0 as? TargetAnnotation,
+        annotation.target.id == targetId
+      {
+        mapView.removeAnnotation(annotation)
+        removeTargetOverlay(targetId: targetId)
+      }
+    }
+  }
+  
+  func removeTargetOverlay(targetId: Int) {
+    guard
+      let overlays = mapView.overlays as? [TargetCircle],
+      let index = overlays.index(where: {$0.targetId == targetId})
```

Comment:
I know that using the `where` is more readable but maybe you can use it like: `overlays.index { $0.targetId == targetId }` Or at just add spaces before and after the brackets 🤷‍♂ Sorry, is my OCD talking 🙈

Reply:
Agree on the spaces 😬 . The trailing closure doesn't work on guard statements (don't ask me why, but the compiler 💥 )

Reply:
@smanini1 what swift version are you using? in Swift 5 this API is `firstIndex`